CppCoreGuidelines.md

```diff
@@ -13138,12 +13138,21 @@ C++ provides better type checking and more notational support.
 It provides better support for high-level programming and often generates faster code.
 
 ##### Example
-
-    char ch = 7;
-    void* pv = &ch;
-    int* pi = pv;   // not C++
-    *pi = 999;      // overwrite sizeof(int) bytes near &ch
-
+    struct dataX{
+        char temperature;
+        char windspeed;
+    }
+    struct dataY{
+        int id;
+        double weight;
+    }
+    int main(){
+        struct dataX x;
+        void* ptrX = &x;
+        struct dataY* = ptrX; // not C++
+        dataY.temperature=20;// overwrite sizeof(int) bytes near &temperature, 
+        //thus corrupting "windspeed" and other data or structure padding
```

Comment:
actually the original example also invoked undefined behaviour in C i.e. casting an char to int. 
I thought it was always allowed to cast from a struct to any other struct in C. 
So i guess there is no valid example in C nor in C++ casting one type to a void pointer and than to an other type. except if you cast a pointer from a type (T) to a char array/ ptr and back to the same type.